src/map/intif.c

```diff
@@ -439,6 +439,97 @@ int intif_request_registry(struct map_session_data *sd, int flag)
 	return 0;
 }
 
+//=================================================================
+// Account Storage
+//-----------------------------------------------------------------
+
+/**
+ * Request the inter-server for a character's storage data.
+ * @packet 0x3010  [out] <account_id>.L
+ * @param  sd      [in]  pointer to session data.
+ */
+void intif_request_account_storage(const struct map_session_data *sd)
+{
+	nullpo_retv(sd);
+
+	/* Check for character server availability */
+	if (intif->CheckForCharServer())
+		return;
+
+	WFIFOHEAD(inter_fd, 6);
+	WFIFOW(inter_fd, 0) = 0x3010;
+	WFIFOL(inter_fd, 2) = sd->status.account_id;
+	WFIFOSET(inter_fd, 6);
+}
+
+/**
+ * Parse the reception of account storage from the inter-server.
+ * @packet 0x3805 [in] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @param  fd     [in] file/socket descriptor.
+ */
+void intif_parse_account_storage(int fd)
+{
+	int account_id = 0, payload_size = 0, storage_count = 0;
+	int i = 0;
+	struct map_session_data *sd = NULL;
+
+	Assert_retv(fd > 0);
+
+	if ((payload_size = RFIFOW(fd, 2) - 8) == 0)
+		return; // no data, nothing to do.
+
+	if ((account_id = RFIFOL(fd, 4)) == 0 || (sd = map->id2sd(account_id)) == NULL) {
+		ShowError("intif_parse_account_storage: Session pointer was null for account id %d!\n", account_id);
+		return;
+	}
+
+	if (sd->storage_received == true) {
+		ShowError("intif_parse_account_storage: Multiple calls from the inter-server received.\n");
+		return;
+	}
+
+	storage_count = (payload_size/sizeof(struct item));
+
+	VECTOR_ENSURE(sd->storage, storage_count, 1);
+
+	for (i = 0; i < storage_count; i++) {
+		const struct item *it = RFIFOP(fd, 8 + i * sizeof(struct item));
+		VECTOR_PUSH(sd->storage, *it);
+	}
+
+	sd->storage_received = true;
+}
+
+/**
+ * Send account storage information for saving.
+ * @packet 0x3011 [out] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @param  sd     [in]  pointer to session data.
+ */
+void intif_send_account_storage(const struct map_session_data *sd)
+{
+	int len = 0, i = 0;
+
+	if (intif->CheckForCharServer())
+		return;
+
+	len = 8 + VECTOR_LENGTH(sd->storage) * sizeof(struct item);
+
+	if (len == 8)
+		return; // nothing to save.
```

Comment:
Hmm, why here though? I understand sending "no data" from char to map to flag it as received but why would the char server need this.